src/model/QMLAdapter.h

```diff
@@ -258,7 +258,7 @@ typename ProviderType::QMLAdapterType *getQMLFrontend(ProviderType *provider)
 }
 
 template<typename ProviderType>
-typename ProviderType::QMLAdapterType *getQMLAdapter(ProviderType *provider)
+typename ProviderType::QMLAdapterType *getQMLAdapter(ProviderType *provider, QQmlEngine* engine = nullptr)
```

Comment:
A better approach is that the QQmlEngine instance used by a sub QMLAdapter is obtained from the parent's QMLAdapter. This is implemented in https://github.com/Pelagicore/facelift/pull/200